### PR TITLE
feat: add `limit.users` option to approval validator

### LINKED ISSUE #509: option to use username list for approval.limit feature
Sometimes there is a need of adding a single username for the approval validation to a certain repo, whereas a common team is used across repos for restricting core approval to certain group of people.

Currently to achieve the same, we'll have to a create a repo specific team. It just adds an overhead for maintenance of multiple teams.

It would be great if we can have something like following - 
```
validate:
      - do: approvals
        min:
          count: 1
        limit:
          team: [ 'some-common-team-shared-across-repos' ]
          users: ['custom-user-1-for-this-specific-repo', 'custom-user-2-for-this-specific-repo']
```

### PR DESCRIPTION
**Pull request by shine2lay**
closes #509

### CODE REVIEW

**Review by vsingal-p — APPROVED** [review]

**Comment by vsingal-p** [comment]
@jusx Can you please review this?

**Review by jusx — APPROVED** [review]
LGTM

File: docs/validators/approval.rst [hunk 1]
```diff
@@ -18,6 +18,7 @@ Approvals
         message: 'Custom message...'
       limit:
         teams: ['org/team_slug'] # when the option is present, only the approvals from the team members will count
+        users: ['user1', 'user2'] # when the option is present, approvals from users in this list will count
```
Comment (jusx) [inline docs/validators/approval.rst:21]: I'm not sure this comment is clear:

1. Does it mean that only these users listed will be summed as the total required approval count?
1. Does it mean that only this subset of users from the team members will count?

I believe the documentation needs work and a comment in this example config is not enough.

I suggest a more detailed description outside of a settings sample.

Comment (vsingal-p) [reply docs/validators/approval.rst:21]: It's option 1 which you mentioned

+1, the documentation can be improved

Comment (shine2lay) [reply docs/validators/approval.rst:21]: added a note on the page to try and clarify it.

**Pull request merged**